modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/model/listener/UserGroupRoleModelListener.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.segments.internal.model.listener;
+
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ClassName;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.model.UserGroupRole;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.segments.SegmentsEntryRetriever;
+import com.liferay.segments.criteria.Criteria;
+import com.liferay.segments.model.SegmentsEntry;
+import com.liferay.segments.service.SegmentsEntryLocalService;
+import com.liferay.segments.service.SegmentsEntryRelLocalService;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Laszlo Pap
+ */
+@Component(immediate = true, service = ModelListener.class)
+public class UserGroupRoleModelListener
+	extends BaseModelListener<UserGroupRole> {
+
+	@Override
+	public void onBeforeRemove(UserGroupRole userGroupRole)
+		throws ModelListenerException {
+
+		try {
+			_deleteUserGroupRoleSegmentData(userGroupRole);
+		}
+		catch (Exception exception) {
+			throw new ModelListenerException(exception);
+		}
+	}
+
+	private void _deleteUserGroupRoleSegmentData(UserGroupRole userGroupRole)
+		throws PortalException {
+
+		long[] userSegmentIDsInGroup =
```

Comment:
This variable name would be much more consistent with the rest of the code as `groupUserSegmentIds`. Note the D is lowercase now, you won't find it uppercase in the rest of the code.

Reply:
Actually it must be `userSegmentsEntryIds`.